interfaces/Portalicious/src/app/pages/user-settings/user-settings.component.ts

```diff
@@ -1,12 +1,182 @@
+
+  newPasswordValidator: ValidatorFn = (
+    control: AbstractControl,
+  ): null | ValidationErrors => {
+    const newPassword = control.get('newPassword');
+    const currentPassword = control.get('currentPassword');
+
+    if (newPassword?.errors?.newEqualsCurrent) {
+      delete newPassword.errors.newEqualsCurrent;
+    }
+
+    if (
+      !newPassword?.hasError('required') &&
+      !newPassword?.hasError('minlength')
+    ) {
+      newPassword?.setErrors(null);
+    }
+
+    let customErrors: { newEqualsCurrent?: true } | null = null;
+
+    const newEqualsCurrentErrorCondition =
+      newPassword?.dirty &&
+      currentPassword?.dirty &&
+      String(newPassword.value) === String(currentPassword.value);
+
+    if (newEqualsCurrentErrorCondition) {
+      customErrors = { newEqualsCurrent: true };
+    }
+
+    const combinedErrors =
+      newPassword?.errors || customErrors
+        ? { ...newPassword?.errors, ...customErrors }
+        : null;
+
+    newPassword?.setErrors(combinedErrors);
+    return combinedErrors;
+  };
+
+  confirmPasswordValidator: ValidatorFn = (
+    control: AbstractControl,
+  ): null | ValidationErrors => {
+    const confirmPassword = control.get('confirmPassword');
+    const newPassword = control.get('newPassword');
+
+    if (confirmPassword?.errors?.confirmDifferentFromNew) {
+      delete confirmPassword.errors.confirmDifferentFromNew;
+    }
+
+    if (!confirmPassword?.hasError('required')) {
+      confirmPassword?.setErrors(null);
+    }
+
+    let customErrors: {
+      confirmDifferentFromNew?: true;
+    } | null = null;
+
+    const confirmDifferentFromNewErrorCondition =
+      confirmPassword?.dirty &&
+      newPassword?.dirty &&
+      confirmPassword.value !== newPassword.value;
+
+    if (confirmDifferentFromNewErrorCondition) {
+      customErrors = { confirmDifferentFromNew: true };
+    }
+
+    const combinedErrors =
+      confirmPassword?.errors || customErrors
+        ? { ...confirmPassword?.errors, ...customErrors }
+        : null;
+
+    confirmPassword?.setErrors(combinedErrors);
+    return combinedErrors;
+  };
+
+  /* eslint-disable-next-line @typescript-eslint/unbound-method */
+  private currentPasswordFormControl = new FormControl('', Validators.required);
+
+  private newPasswordFormControl = new FormControl('', [
+    /* eslint-disable-next-line @typescript-eslint/unbound-method */
+    Validators.required,
+
+    Validators.minLength(8),
+  ]);
+  /* eslint-disable-next-line @typescript-eslint/unbound-method */
+  private confirmPasswordFormControl = new FormControl('', Validators.required);
+
+  changePasswordForm = new FormGroup(
+    {
+      currentPassword: this.currentPasswordFormControl,
+      newPassword: this.newPasswordFormControl,
+      confirmPassword: this.confirmPasswordFormControl,
+    },
+    {
+      validators: [this.newPasswordValidator, this.confirmPasswordValidator],
+    },
+  );
+
+  changePasswordFormSubmitted = signal(false);
```

Comment:
Similar to above, I think we can get this info from the mutation without relying on an extra variable